### PR TITLE
Update google_dragonclaw gpio configuration

### PR DESCRIPTION
**Pull request by niedzwiecki-dawid**
Enable exposed GPIO ports and set default configuration for unused pins as GPIO_DISCONNECTED for saving power consumption.

### CODE REVIEW

File: boards/arm/google_dragonclaw/google_dragonclaw.dts
Comment (fabiobaltieri) [inline boards/arm/google_dragonclaw/google_dragonclaw.dts:97]: what are you trying to achieve with these? normally you'd specify either `GPIO_ACTIVE_HIGH` or `GPIO_ACTIVE_LOW` and a property like `output-high` or `output-low`.

Comment (niedzwiecki-dawid) [reply boards/arm/google_dragonclaw/google_dragonclaw.dts:97]: To save as much power as possible on STM32F412, the unused pins should be configured as analog. This is not a default state of the pins. Flag 0 means GPIO_DISCONNECTED (haven't found a proper dt define), the GPIO driver configures such pin as [analog](https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/gpio/gpio_stm32.c#L84). 
image
this structure explains why analog is the most power efficient mode - it disables Schmitt trigger. Please take a look at [1.2.6](https://www.st.com/resource/en/application_note/an4365-using-stm32f4-mcu-power-modes-with-best-dynamic-efficiency-stmicroelectronics.pdf) for more details.

Comment (duda-patryk) [reply boards/arm/google_dragonclaw/google_dragonclaw.dts:97]: Can we use `#define GPIO_ACTIVE_HIGH        (0 << 0)` instead of 0? I think the GPIO will be disconnected if we don't provide any of `output-high`, `output-low`, `input`.

Comment (niedzwiecki-dawid) [reply boards/arm/google_dragonclaw/google_dragonclaw.dts:97]: Makes sense. I'll add a comment and use `GPIO_ACTIVE_HIGH`.

**Comment by niedzwiecki-dawid** [comment]
Tests won't pass until #65517 is merged.

**Comment by fabiobaltieri** [comment]
> Tests won't pass until #65517 is merged.

That's fine,  you can still update and the we can approve now and we retry and merge once that one is in.

**Comment by niedzwiecki-dawid** [comment]
> > Tests won't pass until #65517 is merged.
> 
> That's fine, you can still update and the we can approve now and we retry and merge once that one is in.

Done, thanks.

File: boards/arm/google_dragonclaw/google_dragonclaw.dts
Comment (fabiobaltieri) [inline boards/arm/google_dragonclaw/google_dragonclaw.dts:95]: maybe mention: without setting the gpio as either input or output, that's the important bit

File: boards/arm/google_dragonclaw/google_dragonclaw.dts
Comment (fabiobaltieri) [inline boards/arm/google_dragonclaw/google_dragonclaw.dts:110]:
```suggestion
		gpios = <13 GPIO_ACTIVE_HIGH>, <14 GPIO_ACTIVE_HIGH>,
			<15 GPIO_ACTIVE_HIGH>;
```

**Review by keith-zephyr — DISMISSED** [review]

File: boards/arm/google_dragonclaw/google_dragonclaw.dts
Comment (niedzwiecki-dawid) [reply boards/arm/google_dragonclaw/google_dragonclaw.dts]: Done. Sorry... I've somehow lost my editor settings.

File: boards/arm/google_dragonclaw/google_dragonclaw.dts
Comment (niedzwiecki-dawid) [reply boards/arm/google_dragonclaw/google_dragonclaw.dts:95]: IMO the `GPIO_DISCONNECTED` part implicates that isn't input or output, but added anyway if you think it can help to understand why it is done this way.

Comment (fabiobaltieri) [reply boards/arm/google_dragonclaw/google_dragonclaw.dts:95]: Thanks, yeah I mean maybe it's a "me" problem but I would expect the default to be "input", there's nothing that says "disconnected" here and this is a STM32 specific combination, I think it's worth being clear.

**Review by fabiobaltieri — APPROVED** [review]

**Comment by niedzwiecki-dawid** [comment]
Ping :-)

**Review by duda-patryk — APPROVED** [review]

**Review by keith-zephyr — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**
